src/amber/websockets/client_socket.cr

```diff
@@ -73,17 +73,20 @@ module Amber
         end
       end
 
-      def session
-        @session ||= @context.session
+      # Authentication and authorization can happen here
+      def on_connect : Bool
```

Comment:
why not `abstract def on_connect : Bool`?

Reply:
also optional, abstract makes it required